activerecord/lib/active_record/delay_touching.rb

```diff
@@ -0,0 +1,119 @@
+require "active_record/delay_touching/state"
+
+module ActiveRecord
+  # = Active Record Delay Touching
+  module DelayTouching # :nodoc:
+    extend ActiveSupport::Concern
+
+    # Override ActiveRecord::Base#touch.  If currently delaying touches, always return
+    # true because there's no way to tell if the write would have failed.
+    def touch(*names)
+      if DelayTouching.delay_touching? && !no_touching?
+        add_to_transaction
+        DelayTouching.add_record(self, names)
+        true
+      else
+        super
+      end
+    end
+
+    # These get added as class methods to ActiveRecord::Base.
+    module ClassMethods
+      # Batches up +touch+ calls for the duration of a transaction.
+      # +after_touch+ callbacks are also delayed until the transaction is committed.
+      #
+      # ==== Examples
+      #
+      #   # Touches Person.first and Person.last in a single database round-trip.
+      #   Person.transaction do
+      #     Person.first.touch
+      #     Person.last.touch
+      #   end
+      #
+      #   # Touches Person.first once, not twice, right before the transaction is committed.
+      #   Person.transaction do
+      #     Person.first.touch
+      #     Person.first.touch
+      #   end
+      #
+      def transaction(*args, &block)
+        super(*args) { DelayTouching.start(*args, &block) }
+      end
+    end
+
+    def self.states
+      Thread.current[:delay_touching_states] ||= []
+    end
+
+    def self.state
+      states.last
+    end
+
+    class << self
```

Comment:
That was an oversight on my part. I wrote the original code while on a team where the preferred style was `self.` and I only used `class << self` in places where it's required, such as the `delegate` declaration.

I'll fix the style to use the Rails convention.